sc-poc/cmd/spacectl/commands/pkg/apply.go

```diff
@@ -0,0 +1,117 @@
+package pkg
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/spacecloud-io/space-cloud/utils"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func newCommandApply() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "apply",
+		Aliases: []string{"a"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client := &http.Client{}
+			creds, err := getCredentials()
+			if err != nil {
+				return err
+			}
+
+			// Login to SpaceCloud
+			if err := login(client, creds); err != nil {
+				return err
+			}
+
+			cfg := ReadConfig()
+			m := make(map[schema.GroupVersionResource][]string)
+
+			// Get all registered sources' GVR
+			path := creds["url"] + "/sc/v1/sources"
+			sourcesGVR, err := listAllSources(client, path)
```

Comment:
Why don't you take the base url as a separate parameter. `path` should be encapsulated in the listAllSources function